streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -384,6 +387,13 @@ public String queryableStoreName() {
     private SuppressedInternal<K> buildSuppress(final Suppressed<? super K> suppress) {
         if (suppress instanceof FinalResultsSuppressionBuilder) {
             final long grace = findAndVerifyWindowGrace(streamsGraphNode);
+            if (grace > Duration.ofHours(1).toMillis()) {
```

Comment:
Should this be `>=` ?

Also, it seems to be very subjective to make the cut-off point 1h -- why not 30 minutes? Or 10 minutes? It might be better to have a way to distinguish between the case for which we run with default grace period or with user specified and log only for the default case?

Having said this, if we can distinguish both cases, we could even throw here... (but this would be a public behavior change and requires a KIP -- don't think we could consider it a bug fix, could we?)

Reply:
The intent is to present a warning if the grace period is long enough to cause usability problems with Suppress. 1 hour is arbitrary; 10 minutes seems too short.

As you point out, an alternative to making a judgement about what is "long enough to cause usability problems" is just to warn if the grace period isn't explicitly set. There is a default grace period, which is `max(24 hours, window size)`. There's no technical issue using this default grace period in conjunction with Suppress. If the default grace period were more punctual, like zero, or 10 minutes, there would be no motivation to warn. Thus, the motivation reduces to "the default grace period is long enough to cause usability problems with suppress". In other words, warning on default grace is just as subjective as warning on any other arbitrary grace period that "seems too long".

Maybe a less subjective approach would be to log an `info` level message simply indicating what amount of time we are configuring suppression to wait. This would still address my concern, the first-15-minutes experience wherein you plug in suppress, hit "go", and nothing happens. Plus, it provides valuable operational information, namely what is the resolved suppression time for the operator: since the value is computed, it would be time-saving to just print out the actual computed value, for our users benefit when debugging the behavior of their applications.

Reply:
Not sure where you are going with this.

> As you point out, an alternative to making a judgement about what is "long enough to cause usability problems" is just to warn if the grace period isn't explicitly set.

So you agree to update the code accordingly or not? If not, why?

> In other words, warning on default grace is just as subjective as warning on any other arbitrary grace period that "seems too long".

I don't think so, because as you mention as "first-15-minutes experience" could be very bad with default settings. Thus, if user want 24h, they can either ignore the warning or overwrite the default with a custom 24h to get rid of the warning (I would expect most users to set a much smaller value anyway, thus I don't think it would have negative impact on UX).

> Maybe a less subjective approach would be to log an `info` level message simply indicating what amount of time we are configuring suppression to wait.

Sounds like a good idea -- I would only WARN if there is something to warn about -- also atm, the code checks for `>`, thus it does not warn with default value and is _also_ warn if a user sets it to an higher value (it seems weird to warn a user about an explicit setting -- there in in generally nothing wrong with a very large grace period)

My suggestion would be, do an INFO log about the actual used value, and maybe do a WARN if default is not overwritten.

Reply:
Ok, the `info` log is what I was proposing, and I've just pushed the change. Can you take another look?

Thanks,
-John